src/pubd/repository.rs

```diff
@@ -822,17 +817,6 @@ impl RrdpServer {
                     } else {
                         let _best_effort_rm = fs::remove_file(path);
                     }
-
-                // clean snapshots no longer referenced in retained notification files
```

Comment:
It was removed because it's relatively complicated with the addition of the random dirs and they would be cleaned up eventually when the delta is out of scope as well. But.. I have now re-implemented this and added a check in `should_publish_files` for this.

Resolving this now, please speak up if you see any remaining issue.